content/posts/2019/08/libplanet-0.5.eng.md

```diff
@@ -0,0 +1,106 @@
+
+[`IRandom.NextDouble()`]: https://github.com/planetarium/libplanet/pull/419
+[official-docs]: https://docs.microsoft.com/en-us/dotnet/api/system.double?view=netstandard-2.0#remarks
+[floating-point-determinism]: https://randomascii.wordpress.com/2013/07/16/floating-point-determinism/
+
+
+Block Action
+-------
+
+In order to keep chains secure, we need miners to mine the blocks, and we need to reward miners to recruit and keep them.
+
+Previously for miners to be rewarded, they had to add a transaction with reward actions to the block each time. But from this version, the new [‘BlockPolicy<T>.BlockAction’] property enables miners to implement a code that rewards them for each block action. 
+
+[`BlockPolicy<T>.BlockAction`]: https://docs.libplanet.io/0.5.0/api/Libplanet.Blockchain.Policies.BlockPolicy-1.html#Libplanet_Blockchain_Policies_BlockPolicy_1_BlockAction
+
+
+`FileStore` Removed
+---------------
+
+Libplanet had been providing an interface called [‘IStore’], [‘FileStore’] (a file-based implementation of IStore), as well as [‘LiteDBStore’] (implementation based on [LiteDB]) since Version [0.4] to simplify the storage layer. Despite the benefits of its simplistic implementation, however, ‘FileStore’ also had the following limitations:
+
+- There were just too many files created because every block, every transaction, and the status of every account block were stored as separate files.
+- With no separate cache or buffer, I/O performance was significantly affected by what physical storage device were used.
+
+As we started using LiteDB, ‘FileStore’ usage rate began to diminish. And so, considering the difficulty of continuously managing ‘FileStore’, we decided not to provide ‘FileStore’ implementation from Version 0.5.
+
+[0.4]: {{< ref "../07/libplanet-0.4" >}}
+[`IStore`]: https://docs.libplanet.io/0.5.0/api/Libplanet.Store.IStore.html
+[`FileStore`]: https://docs.libplanet.io/0.4.0/api/Libplanet.Store.FileStore.html
+[LiteDB]: https://www.litedb.org/
+[`LiteDBStore`]: https://docs.libplanet.io/0.5.0/api/Libplanet.Store.LiteDBStore.html
+
+
+More Detailed Preload Progress
+---------------------------
+
+To carry out the IBD mentioned above, you can call the [‘Swarm<T>.PreloadAsync()’] method. And previously, sending ‘IProgress<BlockDownloadState>’ to this method’s parameter could tell us about the progress of the block download, but it couldn’t tell us about the specific progress from running actions to the final status after the block download (or from a trusted node).
+
+Therefore, the waiting time of this previous method had to be quite boring for players since the loading message only showed either <q>loading</q> or <q>100%</q> and no specific progress of the block download.
+
+However, from Version 0.5, by receiving the ‘IProgress<PreloadState>’ type parameter instead of ‘IProgress<BlockDownloadState>’, this allows us to get a detailed view of the entire preload progress.
```

Comment:
```suggestion
However, from Version 0.5, by receiving the `IProgress<PreloadState>` type parameter instead of `IProgress<BlockDownloadState>`, this allows us to get a detailed view of the entire preload progress.
```